In [4]:
from System.Crawler.crawler import Crawler

crawler = Crawler(downloaded=True)
crawler.get_collection()
crawler.retrieve_all_docs()
print('wtf')

file reut2-002 docs are being processed and extracted to ./.misc/docs/reut2-002
file reut2-004 docs are being processed and extracted to ./.misc/docs/reut2-004
file reut2-008 docs are being processed and extracted to ./.misc/docs/reut2-008
file reut2-007 docs are being processed and extracted to ./.misc/docs/reut2-007
file reut2-015 docs are being processed and extracted to ./.misc/docs/reut2-015
file reut2-003 docs are being processed and extracted to ./.misc/docs/reut2-003
file reut2-006 docs are being processed and extracted to ./.misc/docs/reut2-006
file reut2-016 docs are being processed and extracted to ./.misc/docs/reut2-016
file reut2-012 docs are being processed and extracted to ./.misc/docs/reut2-012
file reut2-020 docs are being processed and extracted to ./.misc/docs/reut2-020
file reut2-000 docs are being processed and extracted to ./.misc/docs/reut2-000
file reut2-011 docs are being processed and extracted to ./.misc/docs/reut2-011
file reut2-010 docs are being processed 

After we have got all the documents lets check the number of distinct words

In [5]:
docs = crawler.retrieve_all_docs()
docs = list(map(lambda x: x[1], docs))
print(docs[0])


 5-MAR-1987 09:07:54.17
earn
uk




 
F
f0986reute
d f BC-JAGUAR-SEES-STRONG-GR   03-05 0115

JAGUAR SEES STRONG GROWTH IN NEW MODEL SALES
    LONDON, March 5 - Jaguar Plc <JAGR.L> is about to sell its
new XJ-6 model on the U.S. And Japanese markets and expects a
strong reception based on its success in the U.K., Chairman Sir
John Egan told a news conference.
    Commenting on an 11 pct growth in 1986 group turnover to
830.4 mln stg and pre-tax profits at 120.8 mln stg, slightly
below 1985's 121.3 mln, Egan said Jaguar aimed at an average
profit growth of 15 pct per year. However, the introduction of
the new model had kept this year's pre-tax profit down.
    Jaguar starts selling XJ-6 in the U.S. In May and plans to
sell 25,000 of its total 47,000 production there in 1987.
    U.S. Sales now account for 65 pct of total turnover,
finance director John Edwards said.
    A U.S. Price for the car has not been set yet, but Edwards
said the relatively high car prices in dollars of 

# Preprocess article

In [1]:
from System.Parser.query_parser import Parser, stemmatiztion, lemmatization
parser = Parser()

In [3]:
print(parser.preprocess('Hello I studied information retrieval this semester, and i quite like the information'))



['$hello$', '$studi$', '$inform$', '$retriev$', '$semest$', '$quit$', '$like$', '$inform$']


In [15]:
import functools 

sets = map(lambda x:set(parser.preprocess(x)) ,docs)
s = functools.reduce(lambda x,y: x|y, sets)
print(len(s))

149278


In [16]:
words = list(s)

print(words[:10])

['$antiwrinkl$', '$initialr$', '$thecibageigi$', '$issatisfi$', '$usabrazilcanada$', '$gasconglomer$', '$developmentstag$', '$ofbeef$', '$branchlast$', '$succ$']


# Size of words is 149278

we notice that the size of words in all the documents after preprocessing is small
therefore we cannot take the whole terms that are present in the documents or
therefore we can take the first 1000 to occur or lets say the most frequent thousand

## Most frequent or the first thousand to arrive

I am not sure which to take because keywords frequency in the documents dont matter as 
their frequency while queries 
I will take the first thousand that occur from random retrieval of documents
